lib/build-plugin.js

```diff
@@ -32,6 +34,10 @@ function buildPlugin(Client, pluginOpts) {
     const client = new Client(opts.clientOptions)
     const concurrency = opts.concurrency || DEFAULT_GET_CONCURRENCY
 
+    // Register client wrapper
+    fastify.decorate('secretClient', new ClientWrapper(Client, fastify, opts.clientOptions))
```

Comment:
Not too sure on this method of exposing, at minimum given there is support for namespacing this would clash with other auth plugins (very bad). Alternatively could think about prefixing with namespace, or giving it a home inside the `secrets` object.

Reply:
then why don't we support namespacing for the client wrapper too?